### PR TITLE
Bug 1783973: Rename rhcos-downloader to machine-os-downloader

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

**Comment by wking** [comment]
/retitle Bug 1783973: Rename rhcos-downloader to machine-os-downloader

No idea how we're going to ratchet these changes in...

File: data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template [hunk 1]
```diff
@@ -6,7 +6,7 @@ set -ex
 IRONIC_IMAGE=$(image_for ironic)
 IRONIC_INSPECTOR_IMAGE=$(image_for ironic-inspector)
 IPA_DOWNLOADER_IMAGE=$(image_for ironic-ipa-downloader)
-COREOS_DOWNLOADER_IMAGE=$(image_for ironic-rhcos-downloader)
+COREOS_DOWNLOADER_IMAGE=$(image_for ironic-machine-os-downloader)
```
Comment (wking) [inline data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template]: To ratchet the installer across, this probably has to be:

```sh
COREOS_DOWNLOADER_IMAGE=$(image_for ironic-machine-os-downloader || image_for ironic-rhcos-downloader)
```

or some such.

Comment (elfosardo) [reply data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template]: done

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1377/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1383/

**Comment by wking** [comment]
Last [images][1] hit a CI-cluster flake:

```
error: error pulling image docker-registry.default.svc:5000/ci-op-ngcdfplk/pipeline@sha256:e081b21dc1a5cae59398e3846bc9b7fea7794223313df29654f9494c4fbba32f: Get https://docker-registry.default.svc:5000/v2/: dial tcp 172.30.113.112:5000: connect: no route to host
```

/retest

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_installer/2821/pull-ci-openshift-installer-master-images/7373

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1384/

**Comment by wking** [comment]
[Previous e2e-aws][1] died on the unrelated [rhbz#1784822][2].

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_installer/2821/pull-ci-openshift-installer-master-e2e-aws/9180
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1784822

**Comment by wking** [comment]
[This e2e-aws][1] is dying with:

```
Failing tests:

[Feature:OpenShiftAuthorization] The default cluster RBAC policy should have correct RBAC rules [Suite:openshift/conformance/parallel]
[Feature:Platform] Managed cluster should have no crashlooping pods in core namespaces over two minutes [Suite:openshift/conformance/parallel]
[Feature:Platform] Service Catalog should check basic usages: OCP-24062, OCP-24049, OCP-15600 [Suite:openshift/conformance/parallel]
```

That's [rhbz#1784822][2], [rhbz#1784837][3], and some API-server crashlooping pods.  All unrelated.  Hopefully we sneak in with the next round of retests.

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_installer/2821/pull-ci-openshift-installer-master-e2e-aws/9184
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1784822
[3]: https://bugzilla.redhat.com/show_bug.cgi?id=1784837

**Comment by wking** [comment]
> ...some API-server crashlooping pods...

Ah, these are [`ContainerStatusUnknown: container could not be located when the pod was terminated`][1].

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1734524

**Comment by wking** [comment]
[Still][1] banging our heads against [rhbz#1734524][2] and [rhbz#1784334][3].  openshift/origin#24323 should help once it lands.

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_installer/2821/pull-ci-openshift-installer-master-e2e-aws/9189
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1734524
[3]: https://bugzilla.redhat.com/show_bug.cgi?id=1784334

**Comment by wking** [comment]
^failed to apply to 4.2 is [fine][1].

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1784505#c1

**Pull request merged**